equinox/nn/_embedding.py

```diff
@@ -176,14 +181,14 @@ def rotate_half(x: Float[Array, "seq_length embedding_size"]):
 
     @staticmethod
     def precompute_freqs_cis(
-        embedding_size: int, end: int, theta: float
+        embedding_size: int, end: Int[ArrayLike, ""], theta: float
     ) -> Complex[Array, "end half_of_embedding_size"]:
         freqs = 1.0 / (
             theta
             ** (jnp.arange(0.0, embedding_size, 2)[jnp.newaxis, :] / embedding_size)
         )
 
-        t = jnp.arange(float(end))
+        t = jnp.arange(end / 1.0)  # promote to float
```

Comment:
(This was also because Pyright kept complaining)

Reply:
I think I preferred `float(end)`, rather than implicitly promoting. Even if that means a pyright-ignore.